crates/data-generator/src/blocks/mod.rs

```diff
@@ -0,0 +1,29 @@
+use std::io::{Error as IoError, Read, Write};
+
+use thiserror::Error;
+
+pub use json::ParseError as JsonParseError;
+
+mod json;
+mod model;
+mod output;
+
+#[derive(Debug, Error)]
+pub enum GenerateError {
+    #[error("I/O operation failed")]
+    Io(#[from] IoError),
+    #[error("JSON parsing failed")]
+    Parse(#[from] JsonParseError),
+}
```

Comment:
This is about as vague as it gets. Replace both messages with `#[error(transparent)]`? @rjacraft/developer